src/test/components/NetworkChart.test.js

```diff
@@ -132,99 +131,148 @@ describe('NetworkChart', function() {
+        },
+      })
     );
 
     expect(getPhaseElementStyles()).toEqual([
       'left: 0px; width: 20px; opacity: 0;',
       'left: 20px; width: 80px; opacity: 0.6666666666666666;',
       'left: 100px; width: 40px; opacity: 1;',
-      'left: 140px; width: 38px; opacity: 0;',
+      'left: 140px; width: 58px; opacity: 0;',
     ]);
   });
 
   it('takes the full width when there is no details in the payload', () => {
     const { getPhaseElementStyles } = setupWithPayload(
-      'Load 101: https://test.mozilla.org',
-      {
-        type: 'Network',
-        URI: 'https://mozilla.org/img/',
-        RedirectURI: 'https://mozilla.org/img/optimized',
-        id: 90001,
-        pri: 20,
-        count: 10,
-        status: 'STATUS_STOP',
+      getNetworkMarkers({
+        uri: 'https://mozilla.org/img/',
+        id: 100,
         startTime: 10,
-        endTime: 99,
-      }
+        // With an endTime at 109, the profile's end time is 110, and so the
+        // profile's length is 100, which gives integer values for test results.
+        endTime: 109,
+      })
     );
+
+    expect(getPhaseElementStyles()).toEqual([
+      'left: 0px; width: 198px; opacity: 1;',
+    ]);
+  });
+
+  it('divides the phases when only the start marker is present', () => {
+    const markerForProfileRange = [
+      'Some Marker',
+      0,
+      // With an endTime at 99, the profile's end time is 100 which gives
+      // integer values for test results.
+      { startTime: 0, endTime: 99 },
+    ];
+
+    const startMarker = getNetworkMarkers({
+      uri: 'https://mozilla.org/img/',
+      id: 100,
+      startTime: 10,
+      endTime: 60,
+    })[0];
+
+    const { getPhaseElementStyles } = setupWithPayload([
+      markerForProfileRange,
+      startMarker,
+    ]);
+
+    expect(getPhaseElementStyles()).toEqual([
+      // The marker goes to the end of the profile range.
+      'left: 0px; width: 180px; opacity: 1;',
+    ]);
+  });
+
+  it('divides the phases when only the end marker is present', () => {
+    const endMarker = getNetworkMarkers({
+      uri: 'https://mozilla.org/img/',
+      id: 100,
+      startTime: 5,
+      fetchStart: 10,
+      // With an endTime at 109, the profile's end time is 110, and so the
+      // profile's length is 100, which gives integer values for test results.
+      endTime: 109,
+      payload: {
+        pri: 20,
+        count: 10,
+        domainLookupStart: 20,
+        domainLookupEnd: 24,
+        connectStart: 25,
+        tcpConnectEnd: 26,
+        secureConnectionStart: 26,
+        connectEnd: 28,
+        requestStart: 30,
+        responseStart: 60,
+        responseEnd: 80,
+      },
+    })[1];
+
+    const { getPhaseElementStyles } = setupWithPayload([endMarker]);
+
     expect(getPhaseElementStyles()).toEqual([
-      'left: 0px; width: 178px; opacity: 1;',
+      'left: 0px; width: 20px; opacity: 0;',
+      'left: 20px; width: 20px; opacity: 0.3333333333333333;',
```

Comment:
Could these floats fail to stringify consistently?

Reply:
yeaaaah I wasn't sure either :-) I'll look if I can make this value a precise value.

Reply:
ah no, I can't change it easily. So my theory is that this is consistent but may change if we change jsdom's version (or react-dom version).